src/Omnipay/WorldpayCGHosted/Message/PurchaseRequest.php

```diff
@@ -203,6 +206,97 @@ public function setPaymentType($value)
         return $this->setParameter('paymentType', $value);
     }
 
+    /**
+     * @return \DateTimeInterface
+     */
+    public function getTokenExpiry()
+    {
+        return $this->getParameter('tokenExpiryDate');
+    }
+
+    /**
+     * @param \DateTimeInterface $dateTime
+     * @return AbstractRequest
+     */
+    public function setTokenExpiry(\DateTimeInterface $dateTime)
+    {
+        return $this->setParameter('tokenExpiryDate', $dateTime);
+    }
+
+    /**
+     * @return string
+     */
+    public function getCreateToken()
+    {
+        return $this->getParameter('createToken');
+    }
+
+    /**
+     * @param string $tokenType
+     * @return AbstractRequest
+     */
+    public function setCreateToken($tokenType)
+    {
+        $allowedValues = [self::TOKEN_TYPE_SHOPPER, self::TOKEN_TYPE_MERCHANT];
+        if (!in_array($tokenType, $allowedValues)) {
+            throw new \InvalidArgumentException(
+                sprintf('Allowed values for createToken are %s', implode(', ', $allowedValues))
+            );
+        }
+        return $this->setParameter('createToken', $tokenType);
+    }
+
+    /**
+     * @return string
+     */
+    public function getTokenEventReference()
+    {
+        return $this->getParameter('tokenEventReference');
+    }
+
+    /**
+     * @param string $reference
+     * @return AbstractRequest
+     */
+    public function setTokenEventReference($reference)
```

Comment:
In the spec on WorldPay it states that this should be

> a unique reference generated by you to track key token events (such as token creation). Maximum 64 characters, and can only contain alphanumeric characters and an underscore (_).

I was unsure about adding some checks in here to ensure the content is to spec, since none of the other code does this I decided to stick with that